packages/orderbook/src/order_provider/base_sra_order_provider.ts

```diff
@@ -0,0 +1,107 @@
+import { APIOrder, HttpClient } from '@0x/connect';
+import { AssetPairsItem, PaginatedCollection, SignedOrder } from '@0x/types';
+
+import { OrderStore } from '../order_store';
+import { AcceptedRejectedOrders, RejectedOrder } from '../types';
+import { utils } from '../utils';
+
+import { BaseOrderProvider } from './base_order_provider';
+
+export abstract class BaseSRAOrderProvider extends BaseOrderProvider {
+    protected readonly _httpClient: HttpClient;
+    protected readonly _networkId?: number;
+    protected readonly _perPage: number;
+
+    /**
+     * This is an internal class for Websocket and Polling Order Providers
+     */
+    constructor(orderStore: OrderStore, httpEndpoint: string, perPage: number, networkId?: number) {
+        super(orderStore);
+        this._httpClient = new HttpClient(httpEndpoint);
+        this._perPage = perPage;
+        this._networkId = networkId;
```

Comment:
If `networkId` is optional, we should also add a default. Otherwise it could be `undefined`.

Reply:
`networkId` is allowed to be undefined in `@0x/connect`. Defaulting (to say 1 for Mainnet) could cause confusing behaviour if for example the SRA endpoint is Kovan only (kovan.sra.0x.org) and networkId 1 is provided as a default.